src/components/shoppingList/shoppingList.tsx

```diff
@@ -32,8 +51,22 @@ const ShoppingList = ({ listId, title, children }: ShoppingListProps) => {
     '--scheme-color-lightest': schemeColorLightest,
   } as CSSProperties
 
-  const toggleDetails = () => {
-    setExpanded(!expanded)
+  const toggleDetails: MouseEventHandler = (e) => {
+    if (
+      deleteRef.current &&
+      deleteRef.current !== e.target &&
+      !deleteRef.current.contains(e.target as Node)
+    ) {
```

Comment:
Ensure the details/list items are not toggled when a user clicks on the delete icon